gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/fullSizeTest/main.cpp

```diff
@@ -0,0 +1,175 @@
+#include <unordered_map>
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+#include <random>
+#include <string>
+#include <map>
+
+using namespace std;
+
+std::string random_string(int length) {
+  // Check if the input length is positive
+  if (length <= 0) {
+    throw std::invalid_argument("Input length must be positive.");
+  }
+
+  // Initialize a random number generator
+  std::random_device rd;
+  std::mt19937 gen(rd());
+
+  // Create a uniform distribution for generating 0 or 1
+  std::uniform_int_distribution<int> distribution(0, 1);
+
+  // Generate the random string
+  std::string result;
+  for (int i = 0; i < length; ++i) {
+    result += std::to_string(distribution(gen));
+  }
+
+  return result;
+}
+std::string extract_sram1(const std::string &init, const std::string &parity) {
```

Comment:
Why is the test creating a copy of the code?
If the actual code changes, this test will not detect the breakage.
Simply include the actual header and test the actual code (https://github.com/os-fpga/Raptor_Tools/blob/74f2ab79214015eaa8b6150f066dfe0dd7485b92/gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h#L343), not a copy of the code.

Reply:
Not a finished work yet, in fact The code in this main was copied back to the .h file as I was making the fix here ...
By the time I make the final check-in things will straiten up.